source/constructs/lib/back-end/back-end-construct.ts

```diff
@@ -49,49 +49,59 @@ export class BackEnd extends Construct {
     });
     props.secretsManagerPolicy.attachToRole(imageHandlerLambdaFunctionRole);
 
-    const imageHandlerLambdaFunctionRolePolicy = new Policy(this, "ImageHandlerFunctionPolicy", {
-      statements: [
-        new PolicyStatement({
-          actions: ["logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents"],
-          resources: [
-            Stack.of(this).formatArn({
-              service: "logs",
-              resource: "log-group",
-              resourceName: "/aws/lambda/*",
-              arnFormat: ArnFormat.COLON_RESOURCE_NAME,
-            }),
-          ],
-        }),
-        new PolicyStatement({
-          actions: ["s3:GetObject", "s3:PutObject", "s3:ListBucket"],
-          resources: [
-            Stack.of(this).formatArn({
-              service: "s3",
-              resource: "*",
-              region: "",
-              account: "",
-            }),
-          ],
-        }),
+    let statements = [
+      new PolicyStatement({
+        actions: ["logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents"],
+        resources: [
+          Stack.of(this).formatArn({
+            service: "logs",
+            resource: "log-group",
+            resourceName: "/aws/lambda/*",
+            arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+          }),
+        ],
+      }),
+      new PolicyStatement({
+        actions: ["s3:GetObject", "s3:PutObject", "s3:ListBucket"],
+        resources: [
+          Stack.of(this).formatArn({
+            service: "s3",
+            resource: "*",
+            region: "",
+            account: "",
+          }),
+        ],
+      }),
+      new PolicyStatement({
+        actions: ["rekognition:DetectFaces", "rekognition:DetectModerationLabels"],
+        resources: ["*"],
+      }),
+    ];
+
+    if (props.s3KmsKeyArn) {
+      statements.push(
         new PolicyStatement({
-          actions: ["rekognition:DetectFaces", "rekognition:DetectModerationLabels"],
-          resources: ["*"],
+          actions: ["kms:decrypt"],
+          resources: [props.s3KmsKeyArn],
         }),
-      ],
+      )
+    }
+
+    const imageHandlerLambdaFunctionRolePolicy = new Policy(this, "ImageHandlerFunctionPolicy", {
+      statements: statements,
     });
 
     addCfnSuppressRules(imageHandlerLambdaFunctionRolePolicy, [
       { id: "W12", reason: "rekognition:DetectFaces requires '*' resources." },
     ]);
     imageHandlerLambdaFunctionRole.attachInlinePolicy(imageHandlerLambdaFunctionRolePolicy);
 
-    const imageHandlerLambdaFunction = new NodejsFunction(this, "ImageHandlerLambdaFunction", {
+    const imageHandlerLambdaFunction = new DockerImageFunction(this, "ImageHandlerLambdaFunction", {
```

Comment:
This is probably the most important change, have to change from a nodejs lambda runtime to a custom docker image runtime so we can build libraries into the environment.